cast-receiver/src/player_manager.ts

```diff
@@ -128,93 +128,94 @@ export default class PlayerManager {
       player.stop();
       player.unload();
     }
-
-    if (this.onPlayerStatusCallback) {
-      this.onPlayerStatusCallback(PlayerStatusEventType.Removed, soundKey);
-    }
   }
 
+  /**
+   * Sets volume for the given player. If the player is currently playing,
+   * it fades the volume instead to provide smooth transition.
+   */
   private setVolume(soundKey: string, volume: number): void {
-    const player = this.players[soundKey];
+    if (this.players.has(soundKey) === false) {
+      return;
+    }
+
+    const player = this.players.get(soundKey);
     if (player.playing()) {
       player.fade(player.volume(), volume, PlayerManager.FADE_VOLUME_DURATION);
     } else {
       player.volume(volume);
     }
   }
 
+  isBuffering(): boolean {
+    return this.bufferingState.size > 0;
+  }
+
   /**
    * Implements the callback to handle messages received from the Sender application.
-   * Based on the action taken, PlayerManager invokes an appropriate callback registered
-   * to it via on* methods.
+   * Based on the action taken, it invokes the status callback with an appropriate
+   * PlayerManagerStatusEvent.
    */
-  handlePlayerEvent(event: PlayerEvent): void {
-    if (event.soundKey in this.players === false) {
-      if (event.action === undefined || event.action === null) {
-        return;
-      }
-
-      if (event.action !== PlayerAction.Create) {
-        // shouldn't be here?
-        return;
-      }
+  handlePlayerEvent(event: PlayerControlEvent): void {
+    event.src.forEach((soundKey: string) => {
```

Comment:
Codacy found an issue: [Arrow function has a complexity of 10. Maximum allowed is 4.](https://app.codacy.com/manual/ashutoshgngwr/noice/pullRequest?prid=6042010)